### PR TITLE
Initialize class/struct variables with default values in platform/

### PR DESCRIPTION
**Pull request by Clawber**
Initializing class/struct member variables in /platform for #52674

### CODE REVIEW

File: platform/android/audio_driver_opensl.h [hunk 1]
```diff
@@ -38,20 +38,20 @@
...
-	bool pause;
+	bool pause = false;
 
-	uint32_t buffer_size;
-	int16_t *buffers[BUFFER_COUNT];
-	int32_t *mixdown_buffer;
-	int last_free;
+	uint32_t buffer_size = 0;
+	int16_t *buffers[BUFFER_COUNT] = {};
+	int32_t *mixdown_buffer nullptr;
```
Comment (akien-mga) [inline platform/android/audio_driver_opensl.h:53]: Missing `=` so it doesn't build.

**Review by akien-mga — COMMENTED** [review]
Some of these structs and classes have constructors which also initialize the members. You should remove the constructor initializations and make sure that the inline initializations you add match what the constructors did. If a constructor ends up empty after this, you can remove it.

**Comment by Clawber** [comment]
I fixed the typo, and removed the duplicate construction initialization. 

Let me know if there are still errors. Thank you very much.

File: platform/server/os_server.h [hunk 2]
```diff
@@ -48,18 +48,18 @@
...
 class OS_Server : public OS_Unix {
-	VisualServer *visual_server;
+	VisualServer *visual_server = nullptr;
 	VideoMode current_videomode;
 	List<String> args;
-	MainLoop *main_loop;
+	MainLoop *main_loop = nullptr;
 
-	bool grab;
+	bool grab = false;
```
Comment (akien-mga) [inline platform/server/os_server.h:56]: You can remove the `OS_Server()` constructor which only initialized this.

File: platform/uwp/joypad_uwp.h [hunk 3]
```diff
@@ -58,12 +58,12 @@ ref class JoypadUWP sealed {
...
+		bool connected = false;
 		ControllerType type;
-		float ff_timestamp;
-		float ff_end_timestamp;
-		bool vibrating;
+		float ff_timestamp = 0;
+		float ff_end_timestamp = 0;
+		bool vibrating = false;
 
 		ControllerDevice() {
```
Comment (akien-mga) [inline platform/uwp/joypad_uwp.h:68]: You should make sure that the values above match the ones used in the constructor - and then remove the constuctor.

File: platform/uwp/os_uwp.h [hunk 4]
```diff
@@ -58,13 +58,13 @@ class OS_UWP : public OS {
...
 		};
 
-		bool alt, shift, control;
-		MessageType type;
-		bool pressed;
-		unsigned int scancode;
-		unsigned int physical_scancode;
-		unsigned int unicode;
-		bool echo;
+		bool alt = false, shift = false, control = false;
```
Comment (akien-mga) [inline platform/uwp/os_uwp.h:61]: I would split those each on their own line.

File: platform/uwp/os_uwp.h [hunk 5]
```diff
@@ -58,13 +58,13 @@ class OS_UWP : public OS {
...
+		MessageType type = KEY_EVENT_MESSAGE;
+		bool pressed = false;
+		unsigned int scancode = 0;
+		unsigned int physical_scancode = 0;
+		unsigned int unicode = 0;
+		bool echo = false;
```
Comment (akien-mga) [inline platform/uwp/os_uwp.h:67]: See `OS_UWP()` constructor and make sure that the values here are in sync, and remove them from the constructor. (The constructor can't be removed in this case but all primitive value assignments can be.)

File: platform/windows/joypad_windows.h [hunk 6]
```diff
@@ -70,43 +70,30 @@ class JoypadWindows {
...
 
 		dinput_gamepad() {
-			id = -1;
-			last_pad = -1;
-			attached = false;
-			confirmed = false;
-
 			for (int i = 0; i < MAX_JOY_BUTTONS; i++)
 				last_buttons[i] = false;
 		}
```
Comment (akien-mga) [inline platform/windows/joypad_windows.h:86]: You can use `bool last_buttons[MAX_JOY_BUTTONS] = {};` and remove the constructor.

**Comment by Calinou** [comment]
@Clawber Bump :slightly_smiling_face: Could you look into [rebasing this pull request](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase against `3.x` then applying the suggestions mentioned above?

**Comment by akien-mga** [comment]
Closing as this is no longer being updated and not ready to merge as is.

**Pull request closed without merging**